### ISSUE TITLE
Java: Represent ctor params in MarkedSource. [phab:D1588]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2009]
_Created by **benjyw** at 2017-04-27 20:58:22_: 

Previously only non-ctor methods had their params represented
in MarkedSource, but there's no reason to exclude them from ctors.

**Comment by kythe** [comment]
_**danielmoy-google** commented inline (context lost during migration) at 2017-04-27 21:05:29_:

Why not 


```
String methodName
if (...) {
  methodName = enclClass...
} else {
  methodName = sym.getS...
}
```

Or can getSimpleName().toString() somehow return null?  I may be confused...

**Comment by kythe** [comment]
_**benjyw** commented inline (context lost during migration) at 2017-04-27 21:17:29_:

No, that works. Changed. It was the other way as residue from a different implementation that relied on fallthrough (which it then turned out is disallowed in the codebase).

**Comment by kythe** [comment]
_**benjyw** wrote at 2017-04-27 21:22:57_:

  - Address code review comment.

**Comment by kythe** [comment]
_**schroederc** wrote at 2017-04-27 22:48:25_:

All tests pass on my local machine.

**Comment by kythe** [comment]
_**schroederc** closed via commit 0fd5cd7957bb0c1b54e0fdd590ca2f9efbd09b72 at 2017-04-28 17:07:32_
